sdk/keyvault/azsecrets/client.go

```diff
@@ -0,0 +1,1011 @@
+func (c *Client) BeginRecoverDeletedSecret(ctx context.Context, secretName string, options *BeginRecoverDeletedSecretOptions) (RecoverDeletedSecretPollerResponse, error) {
+	if options == nil {
+		options = &BeginRecoverDeletedSecretOptions{}
+	}
+	resp, err := c.kvClient.RecoverDeletedSecret(ctx, c.vaultUrl, secretName, options.toGenerated())
+	if err != nil {
+		return RecoverDeletedSecretPollerResponse{}, err
+	}
+
+	getResp, err := c.kvClient.GetSecret(ctx, c.vaultUrl, secretName, "", nil)
+	var httpErr azcore.HTTPResponse
+	if errors.As(err, &httpErr) {
+		if httpErr.RawResponse().StatusCode != http.StatusNotFound {
+			return RecoverDeletedSecretPollerResponse{}, err
+		}
+	}
+
+	b := &beginRecoverPoller{
+		lastResponse:    getResp,
+		secretName:      secretName,
+		client:          c.kvClient,
+		vaultUrl:        c.vaultUrl,
+		recoverResponse: resp,
+		RawResponse:     getResp.RawResponse,
+	}
+
+	return RecoverDeletedSecretPollerResponse{
+		PollUntilDone: b.pollUntilDone,
+		Poller:        b,
+		RawResponse:   getResp.RawResponse,
+	}, nil
+}
+
+// ListDeletedSecrets is the interface for the Client.ListDeletedSecrets operation
+type ListDeletedSecretsPager interface {
+	// PageResponse returns the current ListDeletedSecretPage
+	PageResponse() ListDeletedSecretsPage
+
+	// Err returns true if there is another page of data available, false if not
+	Err() error
+
+	// NextPage returns true if there is another page of data available, false if not
+	NextPage(context.Context) bool
+}
+
+// listDeletedSecretsPager is the pager returned by Client.ListDeletedSecrets
+type listDeletedSecretsPager struct {
+	genPager *internal.KeyVaultClientGetDeletedSecretsPager
+}
+
+// PageResponse returns the current page of results
+func (l *listDeletedSecretsPager) PageResponse() ListDeletedSecretsPage {
+	resp := l.genPager.PageResponse()
+
+	var values []*DeletedSecretItem
+	for _, d := range resp.Value {
+		values = append(values, deletedSecretItemFromGenerated(d))
+	}
+
+	return ListDeletedSecretsPage{
+		RawResponse:    resp.RawResponse,
+		NextLink:       resp.NextLink,
+		DeletedSecrets: values,
+	}
+}
+
+// Err returns an error if the last operation resulted in an error.
+func (l *listDeletedSecretsPager) Err() error {
+	return l.genPager.Err()
+}
+
+// NextPage fetches the next page of results.
+func (l *listDeletedSecretsPager) NextPage(ctx context.Context) bool {
+	return l.genPager.NextPage(ctx)
+}
+
+// ListDeletedSecretsPage holds the data for a single page.
+type ListDeletedSecretsPage struct {
+	// RawResponse contains the underlying HTTP response.
+	RawResponse *http.Response
+
+	// READ-ONLY; The URL to get the next set of deleted secrets.
+	NextLink *string `json:"nextLink,omitempty" azure:"ro"`
+
+	// READ-ONLY; A response message containing a list of the deleted secrets in the vault along with a link to the next page of deleted secrets
+	DeletedSecrets []*DeletedSecretItem `json:"value,omitempty" azure:"ro"`
```

Comment:
Can *DeletedSecretItem -> DeletedSecretItem?

Reply:
I think this should be a ptr for the same reason Jeff mentioned above.